src/main/java/org/mule/extension/http/api/request/builder/SimpleQueryParam.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api.request.builder;
+
+import static java.util.Objects.hash;
+import static org.mule.extension.http.internal.request.LiteralExpressionUtils.resolveLiteralExpression;
+
+import org.mule.runtime.api.el.ExpressionLanguage;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.runtime.parameter.Literal;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents an HTTP Query Parameter
+ *
+ * @since 1.7
+ */
+@Alias("requestQueryParam")
+public class SimpleQueryParam implements KeyValuePair {
+
+  /**
+   * Represents the Key of this HTTP Query Parameter
+   */
+  @Parameter
+  private String key;
```

Comment:
is this used in the context of anything else other than the source? Should this be literal or `@Expression(NOT_SUPPORTED)` ?

Reply:
It's only on the source

Reply:
is this intended to be used somewhere other tahn in the source in the future? Otherwise the name is too generic.

Reply:
It is not intended to be used anywhere else... but which name would be better? It's the key of the Query Param, like in QueryParam that was already implemented. If I changed the name, I would still have to show it as "Key" in Studio